### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by lstrzebinczyk** [kolorahl/waffle_gcs#9]
There has been many attempts to fix some issues with it, but non merged for a while now. If it's dead and we need to make a new one, please tell us.

**Comment by almirsarajcic** [comment]
@lstrzebinczyk, I've merged all the PRs into https://github.com/almirsarajcic/waffle_gcs so you can use my fork.

**Comment by lstrzebinczyk** [comment]
@almirsarajcic Can you make it possible to open issues in your fork? So that we can move the ongoing work there?

**Comment by almirsarajcic** [comment]
Sure, they're open now.

**Comment by dlobo** [comment]
Can we please transfer ownership of the project to almirsarajcic's repo, so it can be updated in hexdocs.pm etc. 

Lots of bug fixes and other issues have been sorted out on that fork, and it would benefit the community a lot to have one working, maintained version. If you can spend a few minutes doing that, it would be highly appreciated

**Comment by almirsarajcic** [comment]
@kolorahl, can you help us in making the fork https://github.com/almirsarajcic/waffle_gcs the main repo?

We'd like to continue working on the package, but at the moment only users who know about the fork can use it and those who come from hex.pm use the stale version.

**Comment by kolorahl** [comment]
Apologies for the massive delay on this stuff. For some reason my notifications were disabled and I wasn't receiving any word about new issues or pull requests coming into this repo. Combine that with the fact I just became a father recently, I sort of just assumed no one was really using this project.

I'm looking over the PRs now and have enabled notifications again. I'd like to start working more on this project again once things calm down at home but if you feel I'm still a roadblock to quick, iterative progress (I won't be offended if you say so), I can absolutely look at transferring the Hex package to your project.

**Comment by almirsarajcic** [comment]
Great news, thank you.

**Comment by dlobo** [comment]
thats awesome news, glad you are back :) 

Would be great to incorporate most/all of the upstream changes and publish that, and then we can collaborate on next steps, what needs to improve etc

**Comment by almirsarajcic** [comment]
There were some new PRs that weren't created in this repo.
Can you take a look at https://github.com/almirsarajcic/waffle_gcs/pulls?q=is%3Apr+is%3Aclosed and maybe create them here as well?

**Comment by dlobo** [comment]
hey @kolorahl  Any timeline on when we can get this version to sync with the version from @almirsarajcic since some of us are using that now

**Comment by kolorahl** [comment]
I'm looking at the closed PRs right now and I'm going to see if I can just cherry-pick them into the master branch and get a new build going. I've been trying to look into current failures to the build and hoping it's just something config-related.

**Comment by dlobo** [comment]
any update on this?

**Comment by kolorahl** [comment]
@almirsarajcic I've contacted Hex.pm for help in transferring the package ownership and what's required on my end to do so. I started the cherry picking process but life seems to be pulling me in a very different direction right now, sadly. I'll get in touch with you if I need anything on your end to complete the transfer. I'll keep this thread open until the transfer is complete so anyone watching this will know when the switch occurs.

**Comment by kolorahl** [comment]
@almirsarajcic Sent an email to your icloud asking for the final piece of info to make the transfer. No rush; respond whenever you've got a moment and I'll see about getting the `waffle_gcs` hex package pushed to your account, then you can feel free to change the source repository and push new versions through your own hex account.

**Comment by kolorahl** [comment]
Sad to have to step back from a project like this, as I still use this at the moment, but better to let more active individuals with the time and dedication keep it alive and under whatever package name is most prolific. It's been an honor to start this package and take it to where I could in the relatively short time I was developing it and it's an honor to have people interested in carrying on the torch as I step aside. Hopefully one day I can come back and maybe submit patches to the "new" `waffle_gcs` source.

**Comment by harrygr** [comment]
Hi @kolorahl, I was wondering what the status of this project is. I've been struggling with @almirsarajcic's fork and unfortunately there seems to be [a problem with getting a Google Auth token](https://github.com/almirsarajcic/waffle_gcs/issues/14).

Am I right in thinking that the "official" latest version (0.1.1) on Hex is yours? 

The latest version of your repo on Hex seems to work, however there is the issue of the dependency resolution (which seems to be fixed in master).

It seems the main blocker to a new version of this being published to Hex is the Travis [build failing due to an invalid/non-existant service account key](https://travis-ci.org/github/kolorahl/waffle_gcs/jobs/739346544#L542) for cloud storage. Clearly the tests rely on hitting GCS so a bucket is needed. Is this something easily fixable?

It would be great to get clarity on which is the right repo to be using for this package.

Many thanks for your work on it so far!

**Comment by kolorahl** [comment]
@harrygr I can try to investigate why the Travis build failed and see if the solution is simple, hopefully just ensuring the appropriate bucket exists is enough, but ultimately I do not have enough free time at the moment to ensure that this repository is maintained and kept up-to-date. It looks like that could change in the near future, so fingers crossed for that.

I can confirm that version 0.1.1  (if that's still on hex.pm) was my last officially-published package from this repo. I'm pretty sure it worked (at least 99% working, certainly in all the common use cases). If you want to stick with that version for now, feel free. I transferred ownership of the repo to @almirsarajcic so any future versions would come from his repo. I may end up either fixing this one (my repo) or look to assisting with the new repo if I get the time to do so.

**TL;DR**

The fork (by @almirsarajcic) is the new official version of this package. Version 0.1.1 is the last published version of the package from my repo, and should be in a working condition for all common use cases. You might be able to use it, at least temporarily, if you are experiencing issues with newer versions.

**Comment by harrygr** [comment]
thanks for the update. Indeed the current version on hex.pm (0.1.1) does work fine. 

The only issue is that hex complains when you install it alongside the current version of waffle due to dependency resolution problems. This is covered in #4 and looks like it's [been fixed](https://github.com/kolorahl/waffle_gcs/commit/31c61623560312bb20a3490d3f70463c9536034d#diff-dfa6f4ed74c90e5d4fda283d547d366586e690387289bcfd473e3fa5f9ace308L40) in master but no release to hex.pm has been made since. I was just thinking that if the CI build could be fixed it would allow a new release that would solve the problem.

FWIW I got around the above issue by adding `override: true` to my waffle dependency.

Thanks again for your work on this. I'll try and have a look the separate problems with @almirsarajcic's fork when I get a chance, but to date it doesn't look like a release has been made from it.

**Comment by almirsarajcic** [comment]
Sorry for the late response, I thought I'd have more time to work on it.

Like Harry said, the issue is with the CI. But besides that, before transferring ownership I've merged a PR that started using new APIs and that messed up many tests. I haven't managed to find time to fix that properly.

@harrygr, if you're interested, you can take a look at that.
